scripts/o3de/o3de/repo.py

```diff
@@ -34,11 +34,30 @@ def process_add_o3de_repo(file_name: str or pathlib.Path,
             logger.error(f'{file_name} failed to load: {str(e)}')
             return 1
 
-        for o3de_object_uris, manifest_json in [(repo_data['engines'], 'engine.json'),
-                                                (repo_data['projects'], 'project.json'),
-                                                (repo_data['gems'], 'gem.json'),
-                                                (repo_data['template'], 'template.json'),
-                                                (repo_data['restricted'], 'restricted.json')]:
+        # A repo may not contain all types of object.
+        manifest_download_list = []
+        try:
+            manifest_download_list.append((repo_data['engines'], 'engine.json'))
+        except Exception:
```

Comment:
The exceptions being caught should be constrained to what the block of code knows how to handle.

In this case KeyError should be caught.
```suggestion
        except KeyError:
```